traffic_ops/app/lib/API/DeliveryService/SslKeys.pm

```diff
@@ -120,12 +120,23 @@ sub view_by_xml_id {
 		if ( !$version ) {
 			$version = 'latest';
 		}
-		$key = "$key-$version";
+		my $ds = $self->db->resultset('Deliveryservice')->search( { xml_id => $xml_id })->single();
+		if (!$ds) {
+			return $self->alert( { Error => " - Could not found delivery service with xml_id=$xml_id!" } );
+		}
+		my $ds_id = $ds->id;
+		my $key = "ds_$ds_id-$version";
 		my $response_container = $self->riak_get( "ssl", $key );
 		my $response = $response_container->{"response"};
-		$response->is_success()
-			? $self->success( decode_json( $response->content ) )
-			: $self->alert( { Error => " - A record for ssl key $key could not be found.  Response was: " . $response->content } );
+		if ($response->is_success()) {
+			my $ssl_keys = decode_json( $response->content );
+			$ssl_keys->{certificate}->{csr} = decode_base64($ssl_keys->{certificate}->{csr}),
```

Comment:
We made this change because it is not compatible with GUI interface. Since this may case regress issue, will revert it back in TC-547.